arbeitszeit_flask/views/pay_means_of_production.py

```diff
@@ -18,19 +18,20 @@ class PayMeansOfProductionView:
     presenter: PayMeansOfProductionPresenter
     template_renderer: UserTemplateRenderer
 
-    def respond_to_get(self, form: PayMeansOfProductionForm):
-        return Response(self._render_template(form), status=200)
+    def respond_to_get(self):
+        return Response(self._render_template(), status=200)
 
     def respond_to_post(self, form: PayMeansOfProductionForm):
         data = self.controller.process_input_data(form)
         if isinstance(data, PayMeansOfProductionController.MalformedInputData):
             self.presenter.present_malformed_data_warnings(data)
-            return Response(self._render_template(form), status=400)
+            return Response(self._render_template(), status=400)
         use_case_response = self.pay_means_of_production(data)
         self.presenter.present(use_case_response)
-        return Response(self._render_template(form), status=200)
+        return Response(self._render_template(), status=200)
 
-    def _render_template(self, form: PayMeansOfProductionForm) -> str:
+    def _render_template(self) -> str:
```

Comment:
With this approach the form is cleared after every request. This introduces a new (minor) problem. If the user makes an error when filling the form and submits it, they get an empty form back. Let's say the user forgot to copy the last character of the plan id. When they sumbit the form, the application will complain that the user input data was incorrect. But instead of keeping the fields like amount and purpose filled in the form will be empty and the user as to fill those fields again.

Consider only clearing the form when the request was handled successfully. Another approach would be to redirect to this instead of responding directly (in line 31).

Reply:
Good point. I went for the first approach.